### PR TITLE
BACK-1479 Increase mempool synchronization frequency

### PR DESCRIPTION
**Pull request by mbasire-ledger**
## What is this about?

The bitcoin/worker handles synchronization of both the blockchain and the mempool. Account synchronization is made by the worker. BTW, It only reacts on events sent by the account manager. The account manager is in charge of the sync frequency.

In order to not stress services used by the worker with useless calls, the worker detects if a new block has been mined since the last account synchronization instead of syncing both blockchain transactions and mempool transactions. 

Changes:
  * Do not try to fetch confirmed transactions if the current block height has not change
  * Do not call interpreter if there is an empty list of txs
  * Use very last block for compute and account manager cursor 
  * The default sync frequency for each account is 10 seconds
 

## Cute picture of lama
lama

### CODE REVIEW

File: account-manager/src/it/resources/application.conf [hunk 1]
```diff
@@ -10,7 +10,7 @@ orchestrator = {
     {
       coin-family = "bitcoin"
       coin = "btc"
-      sync-frequency = 5s
```
Comment (nukky94) [inline account-manager/src/it/resources/application.conf:13]: For it tests it's better to keep at 5s

File: common/src/main/scala/co/ledger/lama/common/models/SyncEvent.scala [hunk 2]
```diff
@@ -96,6 +96,7 @@ case class WorkableEvent[T](
       Instant.now()
     )
 
+  // TODO: newCursor should not be optional
```
Comment (estrauser-ledger) [inline common/src/main/scala/co/ledger/lama/common/models/SyncEvent.scala:99]: todo in other PR ?

File: coins/bitcoin/worker/src/main/scala/co/ledger/lama/bitcoin/worker/services/CursorStateService.scala [hunk 3]
```diff
@@ -1,83 +1,93 @@
...
+
+    private def logUnexpectedError(b: Block, serverError: UnexpectedStatus): IO[Block] = {
+      log.error(
+        s"Error ${serverError.status.code} while calling explorer with block : ${b.hash}"
+      ) *>
+        IO.raiseError(serverError)
+    }
   }
+
+  case class AccountId(id: UUID) extends AnyVal
```
Comment (estrauser-ledger) [inline coins/bitcoin/worker/src/main/scala/co/ledger/lama/bitcoin/worker/services/CursorStateService.scala:92]: There's another AccountId somewhere I believe

**Review by estrauser-ledger — APPROVED** [review]

**Review by nukky94 — APPROVED** [review]

File: common/src/main/scala/co/ledger/lama/common/models/SyncEvent.scala (hunk 2 above)
Comment (mbasire-ledger) [reply common/src/main/scala/co/ledger/lama/common/models/SyncEvent.scala:99]: Yes. We should fix this in a new PR.

File: coins/bitcoin/worker/src/main/scala/co/ledger/lama/bitcoin/worker/services/CursorStateService.scala (hunk 3 above)
Comment (mbasire-ledger) [reply coins/bitcoin/worker/src/main/scala/co/ledger/lama/bitcoin/worker/services/CursorStateService.scala:92]: There are Operation.AccountId, AccountIdentifier and this one. We should at most have 1 AccountId per Service.

**Pull request merged**